### ISSUE TITLE
File not found - nginx docker

### ISSUE THREAD

**Issue by cve** [api-platform/api-platform#646]
Hi,

I just try latest version of api-platform with docker, but I have apache and postgresql installed locally so I changed docker-compose.yml ports accordingly:

````yaml
version: '3.2'

services:
  php:
    image: ${CONTAINER_REGISTRY_BASE}/php
    build:
      context: ./api
      cache_from:
        - ${CONTAINER_REGISTRY_BASE}/php
    depends_on:
      - db
    env_file:
      - ./api/.env
    # Comment out these volumes in production
    volumes:
      - ./api:/srv/api:rw,cached
      # If you develop on Linux, uncomment the following line to use a bind-mounted host directory instead
      # - ./api/var:/srv/api/var:rw

  api:
    image: ${CONTAINER_REGISTRY_BASE}/nginx
    build:
      context: ./api
      dockerfile: Dockerfile.nginx
      cache_from:
        - ${CONTAINER_REGISTRY_BASE}/nginx
    depends_on:
      - php
    volumes:
      - ./api/public:/srv/api/public:ro
    ports:
      - "8080:80"

  cache-proxy:
    image: ${CONTAINER_REGISTRY_BASE}/varnish
    build:
      context: ./api
      dockerfile: Dockerfile.varnish
      cache_from:
        - ${CONTAINER_REGISTRY_BASE}/varnish
    depends_on:
      - api
    # Comment out this volume in production
    volumes:
      - ./api/docker/varnish/conf:/etc/varnish:ro
    ports:
      - "8081:80"

  db:
    # In production, you may want to use a managed database service
    image: postgres:9.6-alpine
    environment:
      - POSTGRES_DB=api
      - POSTGRES_USER=api-platform
      # You should definitely change the password in production
      - POSTGRES_PASSWORD=!ChangeMe!
    volumes:
      - db-data:/var/lib/postgresql/data:rw
      # You may use a bind-mounted host directory instead, so that it is harder to accidentally remove the volume and lose all your data!
      # - ./docker/db/data:/var/lib/postgresql/data:rw
    ports:
      - "6000:5432"

  client:
    # Use a static website hosting service in production
    # See https://github.com/facebookincubator/create-react-app/blob/master/packages/react-scripts/template/README.md#deployment
    image: ${CONTAINER_REGISTRY_BASE}/client
    build:
      context: ./client
      cache_from:
        - ${CONTAINER_REGISTRY_BASE}/client
    env_file:
      - ./client/.env
    volumes:
      - ./client:/usr/src/client:rw,cached
      - /usr/src/client/node_modules
    ports:
      - "81:3000"

  admin:
    # Use a static website hosting service in production
    # See https://github.com/facebookincubator/create-react-app/blob/master/packages/react-scripts/template/README.md#deployment
    image: ${CONTAINER_REGISTRY_BASE}/admin
    build:
      context: ./admin
      cache_from:
        - ${CONTAINER_REGISTRY_BASE}/admin
    env_file:
      - ./admin/.env
    volumes:
      - ./admin:/usr/src/admin:rw,cached
      - /usr/src/admin/node_modules
    ports:
      - "82:3000"

  h2-proxy:
    # Don't use this proxy in prod
    build:
      context: ./h2-proxy
    depends_on:
      - client
      - admin
      - api
      - cache-proxy
    ports:
      - "443:443"
      - "444:444"
      - "8443:8443"
      - "8444:8444"

volumes:
  db-data: {}
```` 
````docker-compose up```` starts normally, but when I open browser with localhost:8080 (my nginx) I see only "File not found", and in docker logs I see:

````bash
api_1          | 2018/04/12 18:36:04 [error] 7#7: *1 FastCGI sent in stderr: "Primary script unknown" while reading response header from upstream, client: 172.24.0.1, server: , request: "GET / HTTP/1.1", upstream: "fastcgi://172.24.0.5:9000", host: "localhost:8080"
api_1          | 172.24.0.1 - - [12/Apr/2018:18:36:04 +0000] "GET / HTTP/1.1" 404 27 "-" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/65.0.3325.181 Safari/537.36" "-"
````

**Comment by teohhanhui** [comment]
Did you figure something out? It'd be good to have more details. Are you using Docker for Windows?

**Comment by teohhanhui** [comment]
Please provide more details to help us help you. :)

**Comment by cve** [comment]
Sorry, but what details am I missing? I'm downloaded latest api-platform to linux with docker, changed ports as described above and it give me "File not found" for localhost:8080

**Comment by teohhanhui** [comment]
Can you check what's in the public directory from inside the `api`(nginx) container?

**Comment by teohhanhui** [comment]
My guess is that it could be a permissions problem. Unfortunately you'll have to figure this one out on your own for now, since I don't think this is something we can troubleshoot easily.

**Comment by gries** [comment]
If you used the zip version instead of the tarball please try to Download it again using the .tar.gz Distribution also could you provide the logs of the php Container ?

**Comment by Mathieudewet** [comment]
Just tried on Debian stretch, workink, can't reproduce.

**Comment by axelmm** [comment]
I had this and similiar issues (not all containers running, not possible 'exec php' ...) using Docker on Proxmox (Debian stretch based) as remote compiler. Sources on Samba share edited from VSCode on Win. 

I tried almost every possible scenarios ... zip, latest, master, tar.gz, git ... and figured out that it was caused by wrong Samba configuration (sources were created with 0700 permission). After fixing smb.conf all containers were created and started properly (from simplest zip).

Errors with Docker on Windows can be permissions related - I had earlier issues with volumes - in fact it was one of many arguments to run remote compiler.
